### PR TITLE
refactor(ENTESB-12218): Replace Swagger with Apicurio model API

### PR DESCRIPTION
**Pull request by christophd**
This is the first step to replace Swagger model API with the Apicurio model API in order to support both version 2.x and 3.x of Open API specification at the same time.

This PR replaces the Swagger models API used in server-api-generator module.

Next step would be to add Open API 3.x support to that module in order to be able to process Open API 3 specifications.

### CODE REVIEW

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   "connectorFactory": "io.syndesis.connector.rest.swagger.ConnectorFactory",
   "dependencies": [
     {
-      "id": "@project.groupId@:@project.artifactId@:@project.version@",
+      "id": "io.syndesis.connector:connector-rest-swagger:@project.version@",
```
Comment (zregvart) [inline app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: hmm odd, works without issues in my IDE, is this a permanent change? Doesn't the `@project.version@` influence the running of tests as well?

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/Oas20ModelParser.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import io.syndesis.server.api.generator.swagger.OpenApiValidationRules;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author Christoph Deppisch
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/Oas20ModelParser.java]: I prefer code without `@author` tags.

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java [hunk 3]
```diff
@@ -366,50 +365,54 @@ static String extendedPropertiesMapSet(final Operation operation) {
         return mapsetJsonTemplate.replace("@enum@", enumJson);
     }
 
-    Optional<Pair<String, Response>> findResponseCode(final Operation operation) {
+    Optional<Pair<String, OasResponse>> findResponseCode(final OasOperation operation) {
+        List<OasResponse> responses = Optional.ofNullable(operation.responses)
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java]: early exit opportunity?

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java [hunk 4]
```diff
@@ -366,50 +365,54 @@ static String extendedPropertiesMapSet(final Operation operation) {
...
     }
 
-    String getResponseCode(final Operation operation) {
+    String getResponseCode(final OasOperation operation) {
         return findResponseCode(operation).get().getKey();
     }
 
-    static List<ConfigurationProperty.PropertyValue> httpStatusList(final Operation operation) {
+    static List<ConfigurationProperty.PropertyValue> httpStatusList(final OasOperation operation) {
+        List<OasResponse> responses = Optional.ofNullable(operation.responses)
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java]: We could also do an early exit here

File: app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json (hunk 1 above)
Comment (christophd) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: It should only work if you have run a Maven build before so Maven already applied the filtering in build output directory. When you do `mvn clean` from time to time you should get the errors.

I think @project.version@ makes sense and the version is not checked in unit tests.

Comment (zregvart) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: I think my IDE is running Maven filtering on resources.

Comment (christophd) [reply app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json]: Lucky you 😃

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseDataShapeGenerator.java [hunk 5]
```diff
@@ -15,47 +15,60 @@
...
+import io.apicurio.datamodels.openapi.v2.models.Oas20Operation;
+import io.apicurio.datamodels.openapi.v2.models.Oas20Response;
 import org.apache.commons.lang3.tuple.Pair;
 
 abstract class BaseDataShapeGenerator implements DataShapeGenerator {
 
-    static Optional<BodyParameter> findBodyParameter(final Operation operation) {
-        final List<Parameter> operationParameters = operation.getParameters();
+    static Optional<OasParameter> findBodyParameter(final OasOperation operation) {
+        final List<OasParameter> operationParameters = Optional.ofNullable(operation.parameters)
```
[the remark this answers is not in the document]
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseDataShapeGenerator.java]: Done

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseDataShapeGenerator.java [hunk 6]
```diff
@@ -15,47 +15,60 @@
...
     }
 
-    static Optional<Response> findResponse(final Operation operation) {
+    static Optional<Oas20Response> findResponse(final OasOperation operation) {
         return findResponseCodeAndSchema(operation).map(Pair::getValue);
     }
 
-    static Optional<Pair<String, Response>> findResponseCodeAndSchema(final Operation operation) {
+    private static Optional<Pair<String, Oas20Response>> findResponseCodeAndSchema(final OasOperation operation) {
+        List<OasResponse> responses = Optional.ofNullable(operation.responses)
```
[the remark this answers is not in the document]
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseDataShapeGenerator.java]: Done

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseOpenApiConnectorGenerator.java [hunk 7]
```diff
@@ -278,60 +277,60 @@ protected final String determineConnectorDescription(final ConnectorTemplate con
...
         }
 
         return title;
     }
 
-    static void addGlobalParameters(final Connector.Builder builder, final Swagger swagger) {
-        final Map<String, Parameter> globalParameters = swagger.getParameters();
+    static void addGlobalParameters(final Connector.Builder builder, final Oas20Document openApiDoc) {
+        final Oas20ParameterDefinitions globalParameters = openApiDoc.parameters;
         if (globalParameters != null) {
```
[the remark this answers is not in the document]
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/BaseOpenApiConnectorGenerator.java]: Done

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java (hunk 3 above)
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java]: Done

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java (hunk 4 above)
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/OpenApiGenerator.java]: Done

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/Oas20ModelParser.java (hunk 2 above)
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/Oas20ModelParser.java]: Yeah it's in my IDE settings from ages ago. Perhaps I need to rethink

**Pull request merged**